miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/NoteRestController.java

```diff
@@ -0,0 +1,176 @@
+      return text;
+    }
+
+    public boolean isInternalOnly() {
+      return internalOnly;
+    }
+
+    public void setInternalOnly(boolean internalOnly) {
+      this.internalOnly = internalOnly;
+    }
+
+    public void setText(String text) {
+      this.text = text;
+    }
+
+  }
+
+  protected static final Logger log = LoggerFactory.getLogger(NoteRestController.class);
+
+  @Autowired
+  private LibraryService libraryService;
+
+  @Autowired
+  private PoolService poolService;
+
+  private final NoteService<?> projectOverviewWrapper = new NoteService<ProjectOverview>() {
+
+    @Override
+    public void addNote(ProjectOverview overview, Note note) throws IOException {
+      projectService.saveProjectOverviewNote(overview, note);
+    }
+
+    @Override
+    public void deleteNote(ProjectOverview overview, Long noteId) throws IOException {
+      projectService.deleteProjectOverviewNote(overview, noteId);
+    }
+
+    @Override
+    public ProjectOverview get(long id) throws IOException {
+      return projectService.getProjectOverviewById(id);
+    }
+  };
+  @Autowired
+  private ProjectService projectService;
+
+  @Autowired
+  private RunService runService;
+
+  @Autowired
+  private SampleService sampleService;
+
+  private <T> void addNote(NoteService<T> service, long entityId, NoteRequest request) throws IOException {
+    T entity = service.get(entityId);
+    Note note = new Note();
+
+    note.setInternalOnly(request.isInternalOnly());
+    note.setText(request.getText());
+    note.setCreationDate(new Date());
+    service.addNote(entity, note);
+  }
+
+  @PostMapping(value = "/{entityType}/{entityId}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void addNote(@PathVariable(name = "entityType") String entityType, @PathVariable(name = "entityId") long entityId,
+      @RequestBody NoteRequest request) throws IOException {
+    addNote(serviceForEntityType(entityType), entityId, request);
+  }
+
+  private <T> void deleteNote(NoteService<T> service, long entityId, long noteId) throws IOException {
+    T entity = service.get(entityId);
+    service.deleteNote(entity, noteId);
+
+  }
+
+  @DeleteMapping(value = "/{entityType}/{entityId}/{noteId}")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  public void deleteNote(@PathVariable(name = "entityType") String entityType, @PathVariable(name = "entityId") long entityId,
+      @PathVariable(name = "noteId") long noteId) throws IOException {
+    deleteNote(serviceForEntityType(entityType), entityId, noteId);
+  }
+
+  private NoteService<?> serviceForEntityType(String entityType) {
+    switch (entityType) {
+    case "sample":
+      return sampleService;
+    case "library":
+      return libraryService;
+    case "pool":
+      return poolService;
+    case "run":
+      return runService;
+    case "projectoverview":
+      return projectOverviewWrapper;
+    default:
+      throw new UnsupportedOperationException("Unknown entity type: " + entityType);
```

Comment:
Should be a RestException with 404 status since this actually indicates a bad url